src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CamundaService.java

```diff
@@ -89,8 +89,48 @@ public void claimTask(String taskId,
 
     }
 
-    public void assignTask(String taskId, String userId) {
+    public void assignTask(String taskId,
+                           AccessControlResponse assignerAccessControlResponse,
+                           List<PermissionTypes> assignerPermissionsRequired,
+                           AccessControlResponse assigneeAccessControlResponse,
+                           List<PermissionTypes> assigneePermissionsRequired) {
+        requireNonNull(
+            assigneeAccessControlResponse.getUserInfo().getUid(),
+            "AssigneeId cannot be null"
+        );
+
+        Map<String, CamundaVariable> variables = performGetVariablesAction(taskId);
+
+        boolean assignerHasAccess = permissionEvaluatorService.hasAccess(
+            variables,
+            assignerAccessControlResponse.getRoleAssignments(),
+            assignerPermissionsRequired
+        );
+
+        if (assignerHasAccess) {
+            boolean assigneeHasAccess = permissionEvaluatorService.hasAccess(
+                variables,
+                assigneeAccessControlResponse.getRoleAssignments(),
+                assigneePermissionsRequired
+            );
+            if (assigneeHasAccess) {
+                performAssignTaskAction(taskId, assigneeAccessControlResponse.getUserInfo().getUid());
+            } else {
+                throw new InsufficientPermissionsException(
+                    String.format("User did not have sufficient permissions to claim task with id: %s", taskId)
+                );
+            }
+        } else {
+            throw new InsufficientPermissionsException(
+                String.format("User did not have sufficient permissions to claim task with id: %s", taskId)
+            );
+        }
+
+    }
 
+    private void performAssignTaskAction(String taskId, String userId) {
+        Map<String, String> body = new ConcurrentHashMap<>();
+        body.put("userId", userId);
         try {
             updateTaskStateTo(taskId, TaskState.ASSIGNED);
```

Comment:
do we want to read in the taskState first to potentially save this call :) ?

Reply:
that's done here -> https://github.com/hmcts/wa-task-management-api/pull/82

Reply:
Ah yes well done, thank you